src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -452,6 +455,20 @@ Error attachToSourceDatabase()
    FilePath existingSdb = sessionDirPath();
    if (existingSdb.exists())
       return sessionDirLock().acquire(sessionLockFilePath(existingSdb));
+   
+   // migrate from 'sdb' to current folder layout if needed
+   bool needsSdbMigration =
+         !sourceDatabaseRoot().exists() &&
+         sdbSourceDatabaseRoot().exists();
+   
+   if (needsSdbMigration)
+   {
+      Error error =
+            sdbSourceDatabaseRoot().copyDirectoryRecursive(sourceDatabaseRoot());
+      
+      if (error)
+         LOG_ERROR(error);
+   }
 
    // check whether we will need to migrate -- ensure we do this only
```

Comment:
As far as I can tell the migration code already in this function is at least 5 years old. Maybe worth removing it so we only have one migration path to maintain?